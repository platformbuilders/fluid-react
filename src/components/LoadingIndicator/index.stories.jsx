import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, select } from '@storybook/addon-knobs';
import markdown from './loadingIndicator.md';
import LoadingIndicator from './index';

const sizeOptions = ['small', 'medium', 'large'];

storiesOf('LoadingIndicator', module)
  .addDecorator(withKnobs)
  .addParameters({
    readme: {
      content: markdown,
    },
    viewport: {
      defaultViewport: '',
    },
  })
  .add('Example', () => (
    <LoadingIndicator size={select('size', sizeOptions, 'small')} />
  ));
