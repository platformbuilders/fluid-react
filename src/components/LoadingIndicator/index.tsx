import React, { FC } from 'react';
import Animation from 'lottie-react-native';
import DefaultAnimation from './animation.json';

const smallSize = { width: 60, height: 60 };
const largeSize = { width: 120, height: 120 };

type Props = {
  large?: boolean;
  animation?: any;
};

const Loading: FC<Props> = ({
  large = false,
  animation = DefaultAnimation,
}) => (
  <Animation
    style={large ? largeSize : smallSize}
    source={animation}
    autoPlay
    loop
  />
);

export default Loading;
