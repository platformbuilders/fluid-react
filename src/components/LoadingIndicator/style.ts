import styled from 'styled-components/native';
import Animation from 'lottie-react-native';
import { moderateScale } from 'react-native-size-matters';
import { getTheme } from '../../utils/helpers';
import { AnimationObject, LoadingVariants } from '../../types';
import LoadingAnimation from '../../assets/animations/loading.json';

export const circularLoading = getTheme('circularLoading');
export const buttonLoading = getTheme('buttonLoading');
export const linearLoading = getTheme('linearLoading');

interface IndicatorProps {
  variant: LoadingVariants;
}

const loadingVariant = (
  props: any,
): string | AnimationObject | { uri: string } => {
  switch (props.variant) {
    case 'button':
      return buttonLoading(props) || LoadingAnimation;
    case 'circular':
      return circularLoading(props) || LoadingAnimation;
    case 'linear':
      return linearLoading(props) || LoadingAnimation;
    default:
      return circularLoading(props) || LoadingAnimation;
  }
};

export const smallSize = {
  width: moderateScale(60),
  height: moderateScale(60),
};

export const largeSize = {
  width: moderateScale(120),
  height: moderateScale(120),
};

export const Indicator = styled(Animation).attrs((props) => ({
  source: loadingVariant(props),
  autoPlay: true,
  loop: true,
}))<IndicatorProps>``;
