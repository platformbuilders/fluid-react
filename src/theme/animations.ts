import LoadingAnimation from '../assets/animations/loading.json';
import { Animations } from '../types';

export default {
  circularLoading: LoadingAnimation,
  buttonLoading: LoadingAnimation,
  linearLoading: LoadingAnimation,
} as Animations;
