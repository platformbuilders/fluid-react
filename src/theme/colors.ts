import { lighten, darken } from 'polished';
import { Colors } from '../utils/types';

const primary = '#C6B09E';
const secondary = '#805281';
const tertiary = '#2A2E39';
const accent = '#547E6A';

const colors: Colors = {
  failure: '#cc0000',
  warning: '#F5B800',
  success: '#357a38',
  transparent: 'transparent',
  disabled: '#EBEBE4',
  primary: {
    light: lighten(0.05, primary),
    main: primary,
    dark: darken(0.12, primary),
    contrast: '#ffffff',
  },
  secondary: {
    light: lighten(0.05, secondary),
    main: secondary,
    dark: darken(0.1, secondary),
    contrast: '#ffffff',
  },
  tertiary: {
    light: lighten(0.15, tertiary),
    main: tertiary,
    dark: darken(0.1, tertiary),
    contrast: '#ffffff',
  },
  accent: {
    light: lighten(0.05, accent),
    main: accent,
    dark: darken(0.05, accent),
    contrast: '#ffffff',
  },
};

export default colors;
