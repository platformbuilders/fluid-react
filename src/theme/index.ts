export { default as colors } from './colors';
export { default as radius } from './radius';
export { default as animations } from './animations';
