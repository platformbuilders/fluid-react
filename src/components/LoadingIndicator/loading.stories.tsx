import { expect } from '@storybook/jest';
import type { Meta, StoryObj } from '@storybook/react';
import { configure, within } from '@storybook/testing-library';
import LoadingIndicator from './index';

configure({ testIdAttribute: 'id' });

//  Mocks
const mockTxt = 'paper-id';

const meta: Meta<typeof LoadingIndicator> = {
  title: 'Components/LoadingIndicator',
  component: LoadingIndicator,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {},
  args: {
    large: true,
    accessibility: mockTxt,
  },
  play: async ({ canvasElement, step }) => {
    const canvas = within(canvasElement);

    await step('LoadingIndicator | Test Render', async () => {
      expect(canvas.getByTestId('loading')).toBeInTheDocument();
    });
  },
};

type Story = StoryObj<typeof LoadingIndicator>;

export const Circular: Story = {
  args: {
    variant: 'circular',
  },
};

export const Linear: Story = {
  args: {
    variant: 'linear',
  },
};

export default meta;
