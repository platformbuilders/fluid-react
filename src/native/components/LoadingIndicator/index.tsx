import React, { FC } from 'react';
import { LoadingType } from '../../types';
import { Indicator, smallSize, largeSize } from './style';

const Loading: FC<LoadingType> = ({
  large = false,
  variant = 'circular',
  ...rest
}) => (
  <Indicator
    testID="loading"
    accessibilitylabel="Aguarde"
    variant={variant}
    style={large ? largeSize : smallSize}
    {...rest}
  />
);

export default Loading;
