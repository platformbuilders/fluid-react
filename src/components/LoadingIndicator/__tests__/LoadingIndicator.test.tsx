import { shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import LoadingIndicator from '..';
import theme from '../../../theme';

describe('<LoadingIndicator />', () => {
  it('should render LoadingIndicator correctly', () => {
    const render = shallow(
      <ThemeProvider theme={theme}>
        <LoadingIndicator />
      </ThemeProvider>,
    );

    expect(render).toMatchSnapshot();
  });

  it('should render LoadingIndicator with large', () => {
    const render = shallow(
      <ThemeProvider theme={theme}>
        <LoadingIndicator large />
      </ThemeProvider>,
    );

    expect(render).toMatchSnapshot();
  });

  it('should render LoadingIndicator with contrast', () => {
    const render = shallow(
      <ThemeProvider theme={theme}>
        <LoadingIndicator contrast />
      </ThemeProvider>,
    );

    expect(render).toMatchSnapshot();
  });

  it('should render LoadingIndicator with variant', () => {
    const render = shallow(
      <ThemeProvider theme={theme}>
        <LoadingIndicator variant="linear" />
      </ThemeProvider>,
    );

    expect(render).toMatchSnapshot();
  });

  it('should render LoadingIndicator with accessibility', () => {
    const render = shallow(
      <ThemeProvider theme={theme}>
        <LoadingIndicator accessibility="" />
      </ThemeProvider>,
    );

    expect(render).toMatchSnapshot();
  });
});
