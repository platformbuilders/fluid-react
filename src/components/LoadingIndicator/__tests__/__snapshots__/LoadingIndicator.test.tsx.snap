// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingIndicator /> should render LoadingIndicator correctly 1`] = `<div />`;

exports[`<LoadingIndicator /> should render LoadingIndicator large contrast with variant 1`] = `<div />`;

exports[`<LoadingIndicator /> should render LoadingIndicator large with variant 1`] = `<div />`;

exports[`<LoadingIndicator /> should render LoadingIndicator with accessibility 1`] = `<div />`;

exports[`<LoadingIndicator /> should render LoadingIndicator with contrast 1`] = `<div />`;

exports[`<LoadingIndicator /> should render LoadingIndicator with large 1`] = `<div />`;

exports[`<LoadingIndicator /> should render LoadingIndicator with variant 1`] = `<div />`;
